### PR TITLE
27 add and remove bin markers on googlemaps

### PR DESCRIPTION
**Pull request by iliasmerigh**
### **In this pull PR I did the following :**

- Added a fake bins database with its tests
- Once the googlemaps fragment is launched it displays all bins with a marker
- When we click on a marker it deletes the bin marker
- When we the "add new bin" button it adds a bin in the current user location
- There are various types of bin : plastic, paper etc... Each one is associated to a color so we can distinguish them in the map.

### **Encountered issues** :

- The bootcamp itself says that the googlemaps API is difficult to test so I didn't manage to achieve a full coverage (for example, click on markers seems impossible to test)

- The bins creation is very long so I decided to focus on the main features that were associated in my issue. In a later spint, I will try to :
• Let the user add custom bin types (for now it automatically selects an arbitrary type)
• Remove a bin in another way than just clicking on its marker
... and maybe some other features

### CODE REVIEW

File: app/src/main/java/com/github/sdp_begreen/begreen/fragments/MapFragment.kt [hunk 1]
```diff
@@ -123,4 +149,66 @@ class MapFragment : Fragment() {
...
+        if (ContextCompat.checkSelfPermission(
+                requireContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED) {
+
+            requestPermissionLauncher.launch(Manifest.permission.ACCESS_FINE_LOCATION)
+
+        } else {
+
+            val bin = Bin(0, BinType.PLASTIC, userLocation!!.latitude, userLocation!!.latitude)
```
Comment (LucaCarroz) [inline app/src/main/java/com/github/sdp_begreen/begreen/fragments/MapFragment.kt:209]:
```suggestion
            val bin = Bin(0, BinType.PLASTIC, userLocation!!.latitude, userLocation!!.longitude)
```
I think there's a little typo here

**Review by LucaCarroz — CHANGES_REQUESTED** [review]
Good addings overall!
Just a few comments made: concerning the use of methods in FirebaseDB, it might be too late to refactor the code to use those methods, but remember the existence of those methods when you store the bin locations on the actual database :)

**Review by LucaCarroz — APPROVED** [review]
Seems good to me!

File: app/src/main/java/com/github/sdp_begreen/begreen/models/Bin.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.github.sdp_begreen.begreen.models
+
+class Bin(val id: Int, val type: BinType, val lat : Double, val long : Double) {
```
Comment (AlSchlo) [inline app/src/main/java/com/github/sdp_begreen/begreen/models/Bin.kt:3]: This can be a data class

**Pull request merged**